rdrf/rdrf/services/rpc/rpc_commands.py

```diff
@@ -1,48 +1,8 @@
 import logging
-logger = logging.getLogger(__name__)
-
-
-def rpc_visibility(request, element):
-    user = request.user
-    if user.can("see", element):
-        return True
-
 
-def rpc_check_notifications(request):
-    from rdrf.models.definition.models import Notification
-    user = request.user
-    results = []
-    notifications = Notification.objects.filter(
-        to_username=user.username, seen=False).order_by('-created')
-    for notification in notifications:
-        results.append({"message": notification.message,
-                        "from_user": notification.from_username, "link": notification.link})
-    return results
-
-
-def rpc_dismiss_notification(request, notification_id):
-    from rdrf.models.definition.models import Notification
-    status = False
-    try:
-        notification = Notification.objects.get(pk=int(notification_id))
-        notification.seen = True
-        notification.save()
-        status = True
-    except Exception as ex:
-        logger.error("could not mark notification with id %s as seen: %s" %
-                     (notification_id, ex))
-    return status
+from django.core.exceptions import PermissionDenied
 
-
-def rpc_fh_patient_is_index(request, patient_id):
-    from registry.patients.models import Patient
-    patient = Patient.objects.get(pk=patient_id)
-    if patient.in_registry("fh"):
-        is_index = patient.get_form_value(
-            "fh", "ClinicalData", "fhDateSection", "CDEIndexOrRelative") != "fh_is_relative"
-        return is_index
-    else:
-        return False
+logger = logging.getLogger(__name__)
 
 
 def rpc_reporting_command(request, query_id, registry_id, command, arg):
```

Comment:
The added code here checks just `get_projection` command.

Any registered users could still call `call_field_data` on any registry to get a list of all the CDEs.
There is no reason to allow that.

Reply:
Added a same registry check. Regarding call_field_data MongoFieldSelector already includes only fields viewable by the user.